community/io/src/test/java/org/neo4j/test/rule/TestDirectory.java

```diff
@@ -246,8 +246,31 @@ private void evaluateClassBaseTestFolder( )
         {
             throw new IllegalStateException(" Test owning class is not defined" );
         }
+        try
+        {
+            testClassBaseFolder = testDataDirectoryOf( fileSystem, owningTest, false );
+        }
+        catch ( IOException e )
+        {
+            throw new RuntimeException( e );
+        }
+    }
+
+    /**
+     * This method can be used outside the context of a Rule as a utility. It will return something like:
+     * {@code component/target/test-data/org.neo4j.location.of.my.test.Clazz/}
+     */
+    public static File testDataDirectoryOf( FileSystemAbstraction fs, Class<?> owningTest, boolean clean )
```

Comment:
It's not a rule factory method and so wanted to call it something else, actually it should be called something very different to not be intermixed with the common rule factory methods, shouldn't it?